README.md

```diff
@@ -4,40 +4,119 @@
+
+## Support for sequential and speech data
+
+Padertorch especially offers support for training with [sequential data](padertorch/ops/sequence) such as:
+- [Masking](padertorch/ops/sequence/mask.py): Calculate a mask which has non-zero entries for non-padded positions in the sequence.
+- [Fragmenting](padertorch/data/fragmenter.py): Fragment a sequence into smaller chunks of same length.
+- [Visualization in tensorboard](padertorch/summary/tbx_utils.py): Prepare spectrograms and speech masks for visualization and audio for playback in tensorboard.
+- [Source separation objectives](padertorch/ops/losses/source_separation.py): Commonly used objectives for speech source separation such as PIT or deep clustering.
 
+# Installation
+
+**Requirements**
+- Python 3
+- torch >= 1.0
+
+```bash
+$ git clone https://github.com/fgnt/padertorch.git
+$ cd padertorch && pip install -e .
+```
+
+# Getting Started
+
+## A Short Explanation of `padertorch.Module` and `padertorch.Model`
+
+You can build your models upon `padertorch.Module` and `padertorch.Model`.
+Both expect a `forward` method which has the same functionality as the `forward` call of  `torch.nn.Module`: It takes data as input, applies some transformations and returns the network output:
+
+```python
+def forward(self, example):
+    x = example['x']
+    out = transform(x)
+    return out
+```
+
+Additionally, `padertorch.Model` expects a `review` method to be implemented which takes the data and output of the `forward` call as inputs from which it computes losses and metrics for logging:
+
+```python
+import torch
+
+def review(self, example, output):
+    # loss computation
+    ce_loss = torch.nn.CrossEntropyLoss()(output, example['x'])
+    # compute additional metrics
+    with torch.no_grad():
+        prediction = torch.argmax(output, dim=1)
+        accuracy = (prediction == example['label']).float().mean()
+    return {
+        'loss': ce_loss,
+        'scalars': {'accuracy': accuracy}
+    }
+```
+
+See [padertorch.summary.tbx_utils.review_dict](padertorch/summary/tbx_utils.py#L213) for how the review dictionary should be constructed.
+For each training step, the trainer calls `forward`, passes its output to `review` and performs a backpropagation step on the loss.
+Typically, the input to the `forward` of a `Module` is a Tensor, while for a `Model`, it is a dictionary which contains additional entries, e.g., labels, which are needed in the `review`.
+This is only a recommendation and there is no restriction for the input type.
+
+While these two methods are mandatory, you are free to add any further methods to your models.
+Since a `Module` does not need a `review` method, it can be used as a component of a `Model`.
+
+## How to Integrate your Data and Model with the Trainer
+
+The trainer works with any kind of iterable data loader, e.g., `torch.utils.data.DataLoader` or `lazy_dataset.Dataset`.
+The `train` method expects a data iterator as input which yields the training data.
```

Comment:
Is iterator the correct word, here? I seem to remember we wanted to use "iterable objects" instead, as the dataset does not exactly iterate.

Reply:
`iterator` is the wrong name, that was the initial name that we used for `lazy_dataset`.
 Here, an idea from my side:
```
The trainer works with any kind of iterable, e.g., `list`,
`lazy_dataset.Dataset` or `torch.utils.data.DataLoader`.
The `train` method expects an iterable as input which
yields examples or minibatches of examples that are
forwarded to the model (i.e., the trainer does not interpret
the yielded entries of the iterable, in our examples it
is always a `dict`).
```

Reply:
Yes, iterable was also used in the previous version of the README. I will update that.